### ISSUE TITLE
Remove historical label on parks with isLegacy

### ISSUE THREAD

**Issue by JLWade** [bcgov/bcparks-ar-admin#250]
As per discussions recent discussions, we have decided to remove the Historical label on parks in A&R as that does not really indicate the park is historical. The historical label will remain on sub-area as that isLegacy flag means it came from historical data, where as the park label does not have the same meaning because the park can still be active.

## Acceptance Criteria

- remove isLegacy flag from Parks/ORCs
- 
- remove historical label on UI for Parks
- 
- keep label and flag for sub-areas where it applied

**Comment by cameronpettit** [comment]
@JLWade just discussing this ticket with @Christopher-walsh22 & I am trying to remember the reasoning behind it. I recall we ran into some issue where a park that had been imported from legacy data (and therefore had the `isLegacy: true` flag) was being reactivated. However, we couldn't perform any actions on the park because the `isLegacy` flag puts a hard lock on the record - you can't make changes to it because then it is no longer 'legacy' data. 

We have the `isLegacy` data flag on every record that was imported from those 2017-2019 historical records. We do this because they weren't created using our system, and therefore they _could_ compromise the integrity of the data in our system. In the event that we found/find something down the line that would/does compromise the system, we have an `isLegacy` flag on all compromised records for quick identification.

All that said, I can't recall what the thought process was that made us decide that we should be removing the `isLegacy` flag from parks. I don't remember what we gain by doing that, and I can only think we lose that `isLegacy` flag that we might need in the event of a data compromise. 

IIRC we can't make new subareas on legacy parks, which needs to be fixed. However, I think this should be done by making subarea creation indifferent to legacy status, not by removing park legacy status. 

Do you remember what the conversation was regarding this? If the removal of the `isLegacy` flag is a business requirement then we can do it, I'd just strongly caution against it.

BTW this has nothing to do with the UI - the `historical` badges should definitely disappear in the UI.

**Comment by JLWade** [comment]
@cameronpettit I think you're right - this came out of the idea that many of theses parks with the "historical" label are not historical (as we are thinking of them now with historical names in DR or repealed) and are actually still active but the sub-area data is the actual "historical" stuff. So for sure remove the historical label on parks regardless. 

I think you're also right in that we we will need to make new sub-areas on any park so that new data can be added to these parks if they want to enter them again. So business requirement is that these parks should not appear historical to the user (because they aren't, so remove that label in the UI), and all parks should be able to have new sub-areas added to them so new data could be added if that park ends up getting reported on again. 

Does that help? @cameronpettit @Christopher-walsh22

**Comment by cameronpettit** [comment]
Yeah and I think the waters are further muddied by mixing the definitions of `historical` and `legacy`. In my head:

- Historical refers to a park/subarea that is no longer a park/subarea, ie it will not be collecting attendance and revenue data.
- Legacy refers to a park/subarea that was brought into the system via that legacy data import from that massive Excel spreadsheet. At the time of importing, the park/subarea was not active but not necessarily historical, meaning it could be reactivated one day and start collecting attendance and revenue data.

By these definitions above, I think the `isLegacy` flag should remain on the parks. I don't think there's such thing as `isHistorical` or similar, but if there is, it should be removed. 

Also by the definitions above, we should allow users to add subareas and collect data for parks even if the `isLegacy` flag is true. 

The fact that parks are keyed off of ORCS which cannot be changed means we have to allow users to edit parks regardless of their legacy status. Its all coming back to me now.

I think we need to look at our API logic and refine the areas where we prevent actions because a park has the `isLegacy` flag, and resist the issue to take the easy path and get rid of the `isLegacy` flag altogether.

**Comment by JLWade** [comment]
That makes sense to me and I like the change to rules around isLegacy field versus removing it. 

Let's go with your above recommendations :)

**Comment by cameronpettit** [comment]
Trying to recall what the series of issues was that lead to the creation of this ticket. IIRC, there were parks brought in from that legacy spreadsheet and were thus given an `isLegacy` flag. These parks were still active, just in a 'dormant' state as they had not collected any attendance and revenue data for some time. 

Now these parks are collection A&R data again. However, in order to record new activity data in a park, the activity must have a `config` object, and that config object must belong to a subarea. The subarea must be created net new, as config objects do not exist on 'historical' subareas - because they will never need to generate new records. The net new subarea must belong to a non-`isLegacy` park for the same reason - you can't edit legacy data. 

SO this massive dependency chain comes down to the fact that the `isLegacy` flag on a park prevents the creation of new subareas, which prevents the creation of new activity configs, which prevents the creation of any new records.

THEREFORE we must allow subareas to be created on `isLegacy` parks. This should be done some other way other than removing the `isLegacy` flag.

If a new subarea can be created for an `isLegacy` park, the `isLegacy` flag remains unchanged, AND all references to 'historical' parks have been eliminated from the UI, then I think the purpose behind this ticket has been achieved.

**Comment by Christopher-walsh22** [comment]
I have been able to create a new subarea on a "isLegacy" park in the dev environment. Not sure if a comment was required but figured it would be worth noting that nothing needed to change in the api to achieve this.

**Comment by manuji** [comment]
Tested on TEST: **Passed**
* Verified that the bubble "Historical" now only displayed for the subareas.
* Passing to PO review.

**Comment by JLWade** [comment]
looks good to me - moving to done
